README.md

```diff
@@ -57,89 +53,223 @@ The following code creates a KNN index with fields my_vector1, my_vector2, my_ve
+This setting enables or disables the circuit breaker feature.  Disabling this setting will keep you at risk of Out of memory as we do not have control on the memory usage for the graphs. The default value is *true*. 
+
+##### knn.memory.circuit_breaker.limit
+This setting indicates the maximum capacity of the cache. When the cache attempts to load in a graph that exceeds this limit, it is forced to evict an entry and *knn.circuit_breaker.triggered *is set to *true.* The default value for this setting is *60% *of the machines total memory outside the Elasticsearch jvm . However, a value in *KB* can be given as well.
 
+###### Example
+If a machine has 100GB RAM. Elasticsearch jvm uses 32GB. Then the default circuit breaker limit is set at 60% of the remaining memory(60% of (100GB -32GB) = 40.8GB)
 
-curl -X PUT "localhost:9200/myindex/_doc/4" -H 'Content-Type: application/json' -d'
+##### knn.circuit_breaker.triggered
+This setting indicates whether  or not the circuit breaker has been triggered. The circuit breaker is triggered if any node in the cluster has had to evict an entry from the cache because the cache’s capacity had been reached. The circuit breaker is untriggered when the size of the entries in the cache goes below *knn.circuit_breaker.unset.percentage. *This can occur when an index is deleted or entries in the cache expire when *knn.cache.item.expiry.enabled* is true. The default value is *False.*
+
+##### knn.circuit_breaker.unset.percentage
+This setting indicates under what percentage of the cache’s total capacity the cache’s current size must be in order to untrigger the circuit breaker. The default value is *75% *of total cache’s capacit*y.*
+
+###### Example
+```
+PUT /_cluster/settings
 {
-"my_vector" : [5.5, 6.5],
-"price":17
+    "persistent" : {
+        "knn.plugin.enabled" : true,
+        "knn.algo_param.index_thread_qty" : 1,
+        "knn.cache.item.expiry.enabled": true,
+        "knn.cache.item.expiry.minutes": 15,
+        "knn.memory.circuit_breaker.enabled" : true,
+        "knn.memory.circuit_breaker.limit" : "55%",
+        "knn.circuit_breaker.unset.percentage": 23
+    }
 }
-'
+```
+
+## Stats
+The KNN Stats API provides information about the current status of the KNN Plugin. The plugin keeps track of both cluster level and node level stats. Cluster level stats have a single value for the entire cluster. Node level stats have a single value for each node in the cluster. A user can filter their query by nodeID and statName in the following way:
+```
+GET /_opendistro/_knn/nodeId1,nodeId2/stats/statName1,statName2
+```
+
+### Cluster Stats
+#### circuit_breaker_triggered
+Indicates whether the circuit breaker is triggered.
+
+### Node Stats
+#### eviction_count
+The number of evictions that have occurred in the guava cache. *note:* explicit evictions that occur because of index deletion are not counted.
+
+#### hit_count
+The number of cache hits that have occurred on the node.
```

Comment:
Can you clarify the meaning here? As in, a user ran a search, and the particular graph the user was trying to search was in memory, so that's... one hit? A hit for each result returned?

Reply:
Yes will clarify